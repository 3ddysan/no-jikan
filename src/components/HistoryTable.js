import React from 'react'
import { library } from '@fortawesome/fontawesome-svg-core'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faCalendarDay, faClock, faCoffee } from '@fortawesome/free-solid-svg-icons'

library.add(faCalendarDay, faClock, faCoffee)

function list(entries) {
    if (entries.length > 0) {
        return entries.map((entry, index) => (
            <tr key={index} className="on-mobile-higher">
                <td>{entry.work}</td>
                <td>{entry.break}</td>
                <td className="has-text-centered">{entry.date}</td>
                <td className="is-hidden-mobile has-text-centered">
                    <a className="button is-light" href="#/">
                        <span className="">Delete</span>
                    </a>
                </td>
            </tr>
        ))
    } else {
        return (
            <tr>
                <td colSpan="4" className='has-text-centered'>
                    There's no data
                </td>
            </tr>
        );
    }
}

const HistoryTable = ({ entries = [] }) => (
    <div className="container">
        <table className="table is-fullwidth">
            <thead>
                <tr>
                    <th>
                        <FontAwesomeIcon icon="clock" />
                        <span className="is-hidden-mobile"> Time</span>
                    </th>
                    <th>
                        <FontAwesomeIcon icon="coffee" />
                        <span className="is-hidden-mobile"> +Break</span>
                    </th>
                    <th className="has-text-centered">
                        <FontAwesomeIcon icon="calendar-day" />
                        <span className="is-hidden-mobile"> Date</span>
                    </th>
                    <th className="has-text-centered is-hidden-mobile">
                        <span>Actions</span>
                    </th>
                </tr>
            </thead>
            <tbody>
                {list(entries)}
            </tbody>
        </table>
    </div>
);

export default HistoryTable;