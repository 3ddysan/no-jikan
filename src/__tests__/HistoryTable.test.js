import React from 'react';
import HistoryTable from 'components/HistoryTable';
import renderer from 'react-test-renderer';

describe('HistoryTable', () => {

  it('renders empty state', () => {
    const tree = renderer
      .create(<HistoryTable entries={[]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders with single ', () => {
    const tree = renderer
      .create(<HistoryTable entries={[{ work: 'work-time', break: 'break-time', date: 'date' }]} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
