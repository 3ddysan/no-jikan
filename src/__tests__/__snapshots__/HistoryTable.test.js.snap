// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryTable renders empty state 1`] = `
<div
  className="container"
>
  <table
    className="table is-fullwidth"
  >
    <thead>
      <tr>
        <th>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-clock fa-w-16 "
            data-icon="clock"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="is-hidden-mobile"
          >
             Time
          </span>
        </th>
        <th>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-coffee fa-w-20 "
            data-icon="coffee"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="is-hidden-mobile"
          >
             +Break
          </span>
        </th>
        <th
          className="has-text-centered"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-calendar-day fa-w-14 "
            data-icon="calendar-day"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h96c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-96zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="is-hidden-mobile"
          >
             Date
          </span>
        </th>
        <th
          className="has-text-centered is-hidden-mobile"
        >
          <span>
            Actions
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="has-text-centered"
          colSpan="4"
        >
          There's no data
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`HistoryTable renders with single  1`] = `
<div
  className="container"
>
  <table
    className="table is-fullwidth"
  >
    <thead>
      <tr>
        <th>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-clock fa-w-16 "
            data-icon="clock"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm57.1 350.1L224.9 294c-3.1-2.3-4.9-5.9-4.9-9.7V116c0-6.6 5.4-12 12-12h48c6.6 0 12 5.4 12 12v137.7l63.5 46.2c5.4 3.9 6.5 11.4 2.6 16.8l-28.2 38.8c-3.9 5.3-11.4 6.5-16.8 2.6z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="is-hidden-mobile"
          >
             Time
          </span>
        </th>
        <th>
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-coffee fa-w-20 "
            data-icon="coffee"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M192 384h192c53 0 96-43 96-96h32c70.6 0 128-57.4 128-128S582.6 32 512 32H120c-13.3 0-24 10.7-24 24v232c0 53 43 96 96 96zM512 96c35.3 0 64 28.7 64 64s-28.7 64-64 64h-32V96h32zm47.7 384H48.3c-47.6 0-61-64-36-64h583.3c25 0 11.8 64-35.9 64z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="is-hidden-mobile"
          >
             +Break
          </span>
        </th>
        <th
          className="has-text-centered"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-calendar-day fa-w-14 "
            data-icon="calendar-day"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V192H0v272zm64-192c0-8.8 7.2-16 16-16h96c8.8 0 16 7.2 16 16v96c0 8.8-7.2 16-16 16H80c-8.8 0-16-7.2-16-16v-96zM400 64h-48V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H160V16c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v48H48C21.5 64 0 85.5 0 112v48h448v-48c0-26.5-21.5-48-48-48z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
          <span
            className="is-hidden-mobile"
          >
             Date
          </span>
        </th>
        <th
          className="has-text-centered is-hidden-mobile"
        >
          <span>
            Actions
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="on-mobile-higher"
      >
        <td>
          work-time
        </td>
        <td>
          break-time
        </td>
        <td
          className="has-text-centered"
        >
          date
        </td>
        <td
          className="is-hidden-mobile has-text-centered"
        >
          <a
            className="button is-light"
            href="#/"
          >
            <span
              className=""
            >
              Delete
            </span>
          </a>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
